sdk/src/client/node_manager/builder.rs

```diff
@@ -87,45 +77,36 @@ impl NodeManagerBuilder {
             url,
             auth: None,
             disabled: false,
+            permanode: false,
         }));
         Ok(self)
     }
 
-    pub(crate) fn with_primary_node(mut self, url: &str, auth: Option<NodeAuth>) -> Result<Self> {
-        let mut url = validate_url(Url::parse(url)?)?;
-        if let Some(auth) = &auth {
-            if let Some((name, password)) = &auth.basic_auth_name_pwd {
-                url.set_username(name)
-                    .map_err(|_| crate::client::Error::UrlAuth("username"))?;
-                url.set_password(Some(password))
-                    .map_err(|_| crate::client::Error::UrlAuth("password"))?;
-            }
+    pub(crate) fn with_primary_nodes(mut self, urls: &[&str]) -> Result<Self> {
```

Comment:
where auth go

Reply:
I have decided to have it only in `with_primary_node()` in the `Node` struct, otherwise the array here gets ugly

Reply:
okie dokie